frontend/public/components/factory/table.tsx

```diff
@@ -463,9 +500,12 @@ const getActiveColumns = (
         }
       }),
     );
-  }
-  if (!_.isEmpty(activeColumns)) {
     columns = columns?.filter((col) => activeColumns.has(col.id) || col.title === '');
+  } else {
+    columns = columns?.filter(
+      (col) =>
+        (activeColumns.has(col.id) && isColumnVisible(col.id, activeColumns)) || col.title === '',
```

Comment:
I don't think the `activeColumns.has(col.id)` check is needed since `isColumnVisible` does that already.

Reply:
@spadgett thanks...remove redundant check.